### DISCUSSION TITLE
Save Model from GUI and use it in Python

### DISCUSSION THREAD

**Discussion by jonathanhoss** [biolab/orange3#5896] [General]
Hello,
I want to save a model which is trained in the GUI and load it afterwards in Python to deploy it as a microservice.
I saved the model as .pkcls File and loaded it into Python. When I try to run the predict function, it gets an ValueError.
How can I access the preprocessing processes that is done in the GUI?
And How do I integrate the preprocessing in my python-script? 

```
import pickle
import Orange
import numpy as np

with open('naive-bayers.pkcls', 'rb') as f:
    mp = pickle.load(f)
    
print(mp)

arguments = np.array([521, 24, 'female', 2, 'own', 'little','little', 3190, 18, 'radio/TV']).reshape(1, -1)

print(arguments)
>>> [['521' '24' 'female' '2' 'own' 'little' 'little' '3190' '18' 'radio/TV']]

print(mp.predict(arguments))
>>> ValueError: Unable to convert array of bytes/strings into decimal numbers with dtype='numeric'

```

**Comment by markotoplak** [comment]
You have to put the data into a learner-compatible `Orange.data.Table` (basically, 3 numpy arrays + a `Domain` describing your columns) to use Orange models. Here are some docs on creating tables from code:
- https://orange3.readthedocs.io/projects/orange-data-mining-library/en/latest/reference/data.table.html

Also, if you create data that is the same as your train input data (before preprocessing), the model's pickle file, which stores the preprocessing, will be able to apply it to the prediction data too. You can try if this works in Orange canvas, with the Predictions widget: pass it the model and the data to predict (not prepreprocessed). The unpickled model, given the same data, will behave the same also used within a script.

**Reply by jonathanhoss** [reply]
Thank you for your fast response. 
In the Orange canvas, I am able to use a pickle file. 
![image](https://user-images.githubusercontent.com/71877338/158790009-9a4904ff-2ea8-4e6c-ac5e-4c8cb336c3cb.png)


In the script, I tried to load the data as an “Orange.data.Table”.
I loaded the same data I used for training: 
`arg = Orange.data.Table('01_DATENSET_Kreditdaten.csv')`. 

If I pass it to the predict function, I get the following error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [31], in <cell line: 1>()
----> 1 mp.predict(arg)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\Orange\classification\naive_bayes.py:85, in NaiveBayesModel.predict(self, X)
     84 def predict(self, X):
---> 85     probs = np.zeros((X.shape[0], self.class_prob.shape[0]))
     86     if self.log_cont_prob is not None:
     87         if sp.issparse(X):

AttributeError: 'Table' object has no attribute 'shape'
```
When I try different models, I get different errors, for KNN for example: `TypeError: len() of unsized object`

**Reply by Gilkovitch** [reply]
it will be nice to have few code lines to show how to run an Orange 3 model in python ...
